### DISCUSSION TITLE
Getting info about existing subjects in stream

### DISCUSSION THREAD

**Discussion by BinaryArchaism** [nats-io/nats-server#3944] [Q&A] [answered]
Hi! 

I have stream with subject `info.*.>` and few publishers (e.g. `info.first_pub.super_info`,  `info.second_pub.super_info`,  etc). So I want to get all info from this `info.*.>` except `info.first_pub.super_info`, how can I do it?

One of solution that came to mind is to subscribe to `info.*.>` and simply ack every msg of unnecessary subject without reading it. But I think this is not really good solution. 
May be I should create subscription for every `info.*.super_info` except unnecessary, but how can I know full name of other subjects? Stream only give info about subject `info.*.>`

Thanks!

**Comment by derekcollison** [comment, marked as answer]
Consumers with multiple subjects are coming in 2.10 that might provide a solution.

**Reply by BinaryArchaism** [reply]
Thank you for answering! But I'll still need info about existing subjects on the stream to collect them and create a multiple subject consumer except subjects I dont need. The stream config shows only wildcard subject, but not a real existing one. The stream state shows number of subjects only
```
"config": {
  "name": "info",
  "subjects": [
      "info.*.>"
  ],
...
"state": {
  ...
  "num_subjects": 6,
  "consumer_count": 1
},
```
So I think it will be great to have multiple subjects consumers but it's not a fully solution for such problems.

**Reply by derekcollison** [reply]
The StreamInfo call has an option to return subject details for all messages. So you will get a map of all subjects with the associated message count.

**Reply by BinaryArchaism** [reply]
Thank you a lot! That helped
